compiler-core/src/javascript/import.rs

```diff
@@ -28,14 +28,15 @@ impl<'a> Imports<'a> {
 
     pub fn register_module(
         &mut self,
-        js_path: String,
+        path: String,
+        is_type: bool,
```

Comment:
I think `is_type` is a little confusing in this context. How about this?

```rust
target: JavaScriptCodegenTarget,
```
```rust
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum JavaScriptCodegenTarget {
    JavaScript,
    TypeScriptDeclarations,
}
```

Reply:
I've changed the thing from being property of an import to an argument for `to_doc` since there should be no type imports in js and no non-type imports in ts, so it is not a property of an import instance. Also less code changes.

As for `enum` - from my perspective it makes sense to introduce the enum if you expect or already know other places where it can be useful too. And instead of `TypeScriptDeclarations` I prefer shorter `TypeScript`.

Do you still want enum after current change? If yes, can you add it (into this PR or after merging)? I feel like you will implement it more properly and consistently with code base than me.

Reply:
An enum with `TypeScriptDeclarations` please, as it's specifically that format and not regular TypeScript.

Reply:
I've updated with the enum. Is that correct usage/placement?